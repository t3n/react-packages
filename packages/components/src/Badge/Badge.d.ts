import React from 'react';
export interface BadgeProps {
    variant: 'inverse' | 'highlight' | 'light';
}
declare const Badge: React.FC<BadgeProps>;
export default Badge;
