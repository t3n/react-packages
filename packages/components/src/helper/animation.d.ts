import { ThemeProps } from '@t3n/theme';
export declare const backgroundAnimation: import("styled-components").FlattenInterpolation<import("styled-components").ThemedStyledProps<ThemeProps, any>>;
