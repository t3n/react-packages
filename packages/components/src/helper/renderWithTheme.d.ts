import 'jest-styled-components';
import React from 'react';
import { RenderOptions } from '@testing-library/react';
import { Theme } from '@t3n/theme';
interface OptionalTheme {
    theme?: Theme;
}
export declare const renderWithTheme: (ui: React.ReactElement, { theme, ...options }: RenderOptions & OptionalTheme) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement>;
export {};
