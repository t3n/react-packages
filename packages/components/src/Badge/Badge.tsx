import React, { ReactNode } from 'react';
import styled from 'styled-components';
import { border, space } from 'styled-system';

import { getColorForBackground, getThemeColor, ThemeProps } from '@t3n/theme';

export interface BadgeProps {
  variant: 'inverse' | 'highlight' | 'light';
  children?: ReactNode;
}

const background = ({
  variant = 'highlight',
  theme,
}: BadgeProps & ThemeProps) => `
  background-color: ${
    variant === 'light'
      ? theme.colors.shades.grey232
      : theme.colors.background[variant]
  };
`;

const StyledBadge = styled.span<BadgeProps>`
  font-size: 0.75rem;
  font-weight: 500;
  text-transform: uppercase;
  ${({ theme }) => border({ theme, borderRadius: '2px' })};
  ${({ theme }) => space({ theme, px: 1, py: '2px' })};
  color: ${(props) =>
    props.variant === 'light'
      ? getThemeColor('text.primary')
      : getColorForBackground(props.variant || 'highlight')};
  ${background};
`;

const Badge: React.FC<BadgeProps> = (props) => <StyledBadge {...props} />;

export default Badge;
