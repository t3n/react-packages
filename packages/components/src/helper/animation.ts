import { css, keyframes } from 'styled-components';

import { ThemeProps } from '@t3n/theme';

const backgroundAnimationKeyframes = ({ theme }: ThemeProps) => keyframes`
  0% {
    background-color: ${theme.colors.shades.grey204}
  }

  50% {
    background-color: ${theme.colors.shades.grey244}
  }

  100% {
    background-color: ${theme.colors.shades.grey204}
  }
`;

// eslint-disable-next-line import/prefer-default-export
export const backgroundAnimation = css`
  animation: ${backgroundAnimationKeyframes} 2s linear infinite;
`;
