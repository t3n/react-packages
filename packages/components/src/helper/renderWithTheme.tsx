/* eslint-disable import/no-extraneous-dependencies */
import 'jest-styled-components';

import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { Theme, theme as originalTheme } from '@t3n/theme';

interface OptionalTheme {
  theme?: Theme;
}

// eslint-disable-next-line import/prefer-default-export
export const renderWithTheme = (
  ui: React.ReactElement,
  { theme, ...options }: RenderOptions & OptionalTheme
) => {
  // merge over default theme

  const mergedTheme = { ...originalTheme, ...theme };

  const Wrapper: React.FC = ({ children }) => (
    <ThemeProvider theme={mergedTheme}>{children}</ThemeProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};
