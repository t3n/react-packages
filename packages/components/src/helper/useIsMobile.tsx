import { Theme } from '@t3n/theme';
import { useCallback, useContext, useEffect, useState } from 'react';
import { ThemeContext } from 'styled-components';

const useIsMobile = (): boolean => {
  const [isMobile, setIsMobile] = useState(false);

  const theme: Theme = useContext(ThemeContext);
  const fontSize = window
    .getComputedStyle(window.document.body)
    .getPropertyValue('font-size');
  const mobileBreakpoint =
    parseInt(theme.breakpoints[1], 10) * parseInt(fontSize, 10);

  const testIsMobile = useCallback(() => {
    if (window.matchMedia(`(max-width: ${mobileBreakpoint}px)`).matches) {
      setIsMobile(true);
    } else {
      setIsMobile(false);
    }
  }, [mobileBreakpoint]);

  useEffect(() => {
    testIsMobile();
    window.addEventListener('resize', testIsMobile);

    return () => {
      window.removeEventListener('resize', testIsMobile);
    };
  }, [testIsMobile]);

  return isMobile;
};

export default useIsMobile;
