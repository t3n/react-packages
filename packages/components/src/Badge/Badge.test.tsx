import React from 'react';

import { renderWithTheme } from '../helper/renderWithTheme';
import Badge from './Badge';

test('Badge matches snapshot', () => {
  const { container } = renderWithTheme(
    <Badge variant="highlight">text</Badge>,
    {},
  );
  expect(container.firstChild).toMatchSnapshot();
});
