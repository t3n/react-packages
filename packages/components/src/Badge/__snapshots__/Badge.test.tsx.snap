// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge matches snapshot 1`] = `
.c0 {
  font-size: 0.75rem;
  font-weight: 500;
  border-radius: 2px;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 2px;
  padding-bottom: 2px;
  color: #fff;
  background-color: #f9423a;
}

<span
  class="c0"
>
  text
</span>
`;
